bob-access/src/extractor.rs

```diff
@@ -40,14 +53,16 @@ impl<T: Extractor> ExtractorExt for T {
 
     fn extract_basic(&self) -> Result<Credentials, Error> {
         let mut builder = prepare_builder(self)?;
-        match (self.get_header("username")?, self.get_header("password")?) {
-            (Some(username), Some(password)) => {
+        let auth_header = self.get_header("authorization")?;
+        if let Some(auth_header) = auth_header {
+            let mut parts = auth_header.split_whitespace();
+            if let (Some("Basic"), Some(credentials)) = (parts.next(), parts.next()) {
```

Comment:
Unfortunately, authorization scheme should be case-insensitive according to RFC (https://www.rfc-editor.org/rfc/rfc2617). So you should compare with `Basic` in case insensitive way. Please, don't use `to_lowercase` conversion in comparison, because this logic is on the hot path. It is better to find (or write) a function, that do case-insensitive comparison in letter-by-letter way without string allocation